### ISSUE TITLE
Automatically run code blocks when building documentation in CI

### ISSUE THREAD

**Issue by nspope** [popsim-consortium/stdpopsim#1390]
It's a bit tedious to go through every code block in the tutorial, check that these work, and fix them. I wonder if they could be automatically run as part of the CI. There would probably need to be some sort of caching, as the example simulations take ~10 minutes to run in some cases.

**Comment by jeromekelleher** [comment]
This is one of the reasons we moved to JupyterBook for msprime/tskit etc @nspope. I don't think there is a way to run this stuff well without it - but conversion is a significant task, and examples would need to be rejigged to run in much less time (since they'll be run every time the docs are built)

**Comment by nspope** [comment]
Makes sense! I don't think there's a rush as the examples are up-to-date now, and probably won't drift too much before 0.2.1 or whatever is next.
